packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -166,7 +166,9 @@ mixin CupertinoRouteTransitionMixin<T> on PageRoute<T> {
   @override
   bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
     // Don't perform outgoing animation if the next route is a fullscreen dialog.
```

Comment:
I would update this comment to reflect the new logic here. And/or, pull out some of the logic into a boolean variable with a descriptive name.